apps/movies/src/movies.controller.ts

```diff
@@ -28,8 +31,25 @@ export class MoviesController {
   }
 
   @Get()
-  findAll() {
-    return this.moviesService.findAll();
+  async getPaginatedMovies(
+    @Query('page', new DefaultValuePipe(1), ParseIntPipe) page = 1,
+    @Query('limit', new DefaultValuePipe(20), ParseIntPipe) limit = 20
+  ): Promise<PaginationResponseDto<MovieResponseDto[]>> {
```

Comment:
```suggestion
  ): Promise<Pagination<MovieResponseDto[]>> {
```

you can use import { Pagination } from 'nestjs-typeorm-paginate';

Reply:
Actually let's create a shared interfaces file in lib folder where you can create your own Pagination interface which will extends Pagination from 'nestjs-typeorm-paginate' and please do the same for all types/interfaces from nestjs-typeorm-paginate
We do so to have an ability quickly replace nestjs-typeorm-paginate logic by any other approach/lib